### DISCUSSION TITLE
对于pipewire的更新和一些系统基础包更新的讨论

### DISCUSSION THREAD

**Discussion by xzl01** [linuxdeepin/developer-center#3898] [Developer Discussion | 开发者讨论]
## 我期望升级deepin v23系统中的pipewire相关组件：
- [ ] pipewire = 0.3.65

### 新增
- [ ] libcamera = 0.0.3-6
- [ ] freeaptx = 0.0.1-2
- [ ] liblc3 = 1.0.1
- [ ] wireplumber = 0.4.13

### 同时期望升级相关基础构建软件包：

- [ ] catch2 = 2.13.10-1
- [ ] fmtlib = 9.1.0
- [ ] libhowardhinnant-date = 3.0.1
- [ ] sndio = 1.9.0
- [ ] spdlog = 1.10

### 目前已知影响：
如果安装pipewire会使得pulseaudio无法正常工作，故以后系统不会预装pipewire

**Comment by Zeno-sole** [comment]
升级的目的是什么 @xzl01 ?

**Reply by xzl01** [reply]
目的是支持pipewire的蓝牙高清音频和对waybar构建基础的支持

**Comment by Decodetalkers** [comment]
不是啦，炸pulseaudio是不正常的事情。理论上不应该炸的

**Reply by xzl01** [reply]
我之前在arch上也测试过，在安装pipewire的时候 pipewire会期望使用pipewire-pulseauido去取代pulseauido后端，导致音频无法正常播放

**Reply by zccrs** [reply]
> 我之前在arch上也测试过，在安装pipewire的时候 pipewire会期望使用pipewire-pulseauido去取代pulseauido后端，导致音频无法正常播放

是不需要默认安装，不过装了之后，pipewire-pulseaudio 也应该是个能用的 pulseaudio 后端

**Comment by BLumia** [comment]
此组件本身在由谁维护，多媒体组？是的话，也和多媒体组沟通下？如果不是的话，你打算接手维护此软件包以及其相关依赖的后续更新吗？

**Reply by xzl01** [reply]
现在看来 多媒体组是不存在的，肯定后续是我们接手维护了

**Comment by YukariChiba** [comment]
`pipewire` 能够接管大部分使用 `alsa` `pulseaudio` `jack` 等音频服务器的程序，更换后这些程序应当正常工作。

另外，wayland 下的许多屏幕共享应用依赖于 pipewire，个人建议后续将默认的音频服务器迁移到 pipewire。

**Reply by zccrs** [reply]
先把包打出来，能用。至于什么时候换成默认，可以另外再说。我们甚至可以做成可切换的，在系统设置里留出选项。但是这一切的前提都是要先有pipewire才行，现在是完全没有支持 pipewire。

**Comment by BLumia** [comment]
关联：

- https://github.com/linuxdeepin/developer-center/discussions/4000
- https://github.com/linuxdeepin/developer-center/discussions/5566

**Comment by YukariChiba** [comment]
`pipewire` 预计可在 Beta2 进入支持状态，DDE 现已对 `pipewire` 进行适配。后续可以提供切换方案。

**Reply by talesam** [reply]
I found a bug, Deepin comes with a pipeware and pulseaudio option to choose from, if you choose pulseaudio, the system crashes after logoff. The stranger who doesn't have pulseaudio installed and still appears to choose from deepin. Both conflict with each other, or install one or the other.
